pkg/accountwallet/commands.go

```diff
@@ -46,83 +54,295 @@ func (a *AccountWallet) createAccountImplicitly(ctx context.Context, params *Cre
 
-	//signedTx, err := txBuilder.Build(a.genesisHdWallet.AddressSigner())
+	implicitAccountHandler := wallet.NewEd25519Account(iotago.AccountIDFromOutputID(implicitAccountOutput.OutputID), privateKey)
+	blkID, err := a.PostWithBlock(ctx, a.client, signedTx, implicitAccountHandler, congestionResp, issuerResp, version)
+	if err != nil {
+		log.Errorf("Failed to post account with block: %s", err)
 
-	//accountID := a.registerAccount(params.Alias, implicitAccountOutput.OutputID, a.latestUsedIndex, privateKey)
+		return iotago.EmptyAccountID, ierrors.Wrap(err, "failed to post transaction")
+	}
+
+	accountID := iotago.AccountIDFromOutputID(implicitAccountOutput.OutputID)
+	err = a.checkAccountStatus(ctx, blkID, accountID)
+	if err != nil {
+		return iotago.EmptyAccountID, ierrors.Wrap(err, "failure in account creation")
+	}
+
+	a.registerAccount(params.Alias, implicitAccountOutput.OutputID, accAddrIndex, accPrivateKey)
 
-	//fmt.Printf("Created account %s with %d tokens\n", accountID.ToHex(), params.Amount)
-	return iotago.EmptyAccountID, nil
+	return accountID, nil
 }
 
-func (a *AccountWallet) createAccountWithFaucet(_ *CreateAccountParams) (iotago.AccountID, error) {
-	log.Debug("Not implemented yet")
-	//faucetOutput := a.getFaucetOutput()
-	//
-	//accountOutputBuilder := builder.NewAccountOutputBuilder(accAddr, accAddr, implicitAccountOutput.Balance).
-	//	Mana(implicitAccountOutput.OutputStruct.StoredMana()).
-	//	AccountID(iotago.AccountIDFromOutputID(implicitAccountOutput.OutputID))
+func (a *AccountWallet) requestEnoughFundsForAccountCreation(ctx context.Context, currentMana iotago.Mana) ([]*models.Output, error) {
+	requiredMana, err := a.estimateMinimumRequiredMana(ctx, 1, 0, false, true)
+	if err != nil {
+		return nil, ierrors.Wrap(err, "failed to estimate number of faucet requests to cover minimum required mana")
+	}
+	log.Debugf("Mana required for account creation: %d, requesting additional mana from the faucet", requiredMana)
+	if currentMana >= requiredMana {
+		return []*models.Output{}, nil
+	}
+	additionalBasicInputs, err := a.RequestManaFromTheFaucet(ctx, requiredMana)
+	if err != nil {
+		return nil, ierrors.Wrap(err, "failed to request mana from the faucet")
+	}
+	log.Debugf("successfully requested %d new outputs from the faucet", len(additionalBasicInputs))
+
+	return additionalBasicInputs, nil
+}
+
+func (a *AccountWallet) RequestManaFromTheFaucet(ctx context.Context, minManaAmount iotago.Mana) ([]*models.Output, error) {
+	if minManaAmount/a.faucet.RequestManaAmount > MaxFaucetManaRequests {
+		return nil, ierrors.Errorf("required mana is too large, needs more than %d faucet requests", MaxFaucetManaRequests)
+	}
+
+	outputs := make([]*models.Output, 0)
+	wg := sync.WaitGroup{}
+	// if there is not enough mana to pay for account creation, request mana from the faucet
+	for requested := iotago.Mana(0); requested < minManaAmount; requested += a.faucet.RequestManaAmount {
+		wg.Add(1)
+		go func(requested iotago.Mana) {
```

Comment:
```suggestion
		go func(ctx context.Context, requested iotago.Mana) {
```
Let's also have context here.